src/Tools/Source/RunTests/ProcessUtil.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Management;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace RunTests
+{
+    internal static class ProcessUtil
+    {
+        internal static int? GetParentProcessId(Process p)
```

Comment:
TryGet... to illustrate the "this might fail" nature? (still OK with the nullable return.)